lang/intrinsic.md

```diff
@@ -0,0 +1,49 @@
+# Intrinsics
+
+```rust
+pub enum Intrinsic {
+    Exit,
+    PrintStdout,
+    PrintStderr,
+}
+
+impl<M: Memory> Machine<M> {
+    fn eval_intrinsic(
+        &mut self,
+        intrinsic: Intrinsic,
+        arguments: List<Value<M>>,
+        ret: Option<Place<M>>,
+        next_block: Option<BbName>,
+    ) -> NdResult {
+        match intrinsic {
+            Intrinsic::Exit => throw_machine_stop!("0"),
+            Intrinsic::PrintStdout => self.eval_print(&mut std::io::stdout(), arguments, ret, next_block)?,
+            Intrinsic::PrintStderr => self.eval_print(&mut std::io::stderr(), arguments, ret, next_block)?,
+        }
+    }
+
+    fn eval_print(
+        &mut self,
+        stream: &mut impl std::io::Write,
+        arguments: List<Value<M>>,
+        _ret: Option<Place<M>>,
```

Comment:
Do we need to assign `()` to `ret`?
`print` is definitely not "never-returning".

Reply:
As of this PR `print` would accept both a `Some` and a `None` `ret`.
I think we would decide on which is correct before merging this.
I'm tending towards `Some`, as `print` behaves like a `-> ()` function, which also has a return place.

Reply:
I think accepting both is fine, and you don't need to write anything for a ZST return value. A copy of 0 bytes is a NOP anyway.